scapy/utils.py

```diff
@@ -429,17 +429,21 @@ class ContextManagerSubprocess(object):
     >>>     subprocess.Popen(["unknown_command"])
 
     """
-    def __init__(self, name):
+    def __init__(self, name, prog):
         self.name = name
+        self.prog = prog
 
     def __enter__(self):
         pass
 
     def __exit__(self, exc_type, exc_value, traceback):
-        if exc_type == OSError:
-            msg = "%s: executing %r failed"
-            log_runtime.error(msg, self.name, conf.prog.wireshark, exc_info=1)
-            return True  # Suppress the exception
+        if isinstance(exc_value, (OSError, TypeError)):
+            msg = "%s: executing %r failed" % (self.name, self.prog) if self.prog else "Could not execute %s, is it installed ?" % self.name
+            if conf.interactive:
```

Comment:
Hmmm. I was thinking the other way around: in interactive mode, you write a warning log. In non-interactive mode, you raise the exception. That's what has been done in other places by @guedou, and that makes sense: when you are programming, you handle exceptions. When you are using Scapy, you read the warnings.